skyfield/timelib.py

```diff
@@ -218,11 +218,13 @@ def __len__(self):
         return self.shape[0]
 
     def __repr__(self):
-
-        if getattr(self.tt, 'shape', ()):
+        if getattr(self.tt, 'shape', ()) and self.tt.shape == (0,):
```

Comment:
1. In the usual case that the array is non-empty, this new code performs 3 lookups of `.shape` — could you start out with a statement `shape = …` to run `getattr` only once, then re-use the value each time you need it?

2. Would it make sense to test `self.tt.size == 0` instead, so that empty n-dimensional arrays get the `[]`, and not just 1-dimensional empty arrays?

(And so forth with the other instances of this pattern.)

Reply:
I like the idea of testing for the size attribute, which would work great for Time objects. However, if Angle, Distance, and Velocity objects are initialized with a python float, then self.radians, self.au, and self. au_per_d become python floats, which don't have the size attribute. This is the situation I was trying to avoid by testing if the `shape` attribute exists before calling it. For these objects, can floats be converted to numpy float64's before being used to initialize the object? Perhaps by adding another case to the `_to_array` function? If that's possible I think that's the best solution.

Reply:
Alas, my explanation was too short! I was only trying to suggest that the second term of your `and` be adjusted — you are quite right that you need the initial check for shape, protected by `getattr`. I was thinking of:
```
        shape = getattr(self.tt, 'shape', ())
        if shape and self.tt.size == 0:
```
That would protect against the case of a plain float, I think?